tutorials/scripting/c_sharp/c_sharp_collections.rst

```diff
@@ -0,0 +1,233 @@
+.. _doc_c_sharp_collections:
+
+C# collections
+==============
+
+Choose a collection
+-------------------
+
+The .NET base class library contains multiple collection types that can be
+used to store and manipulate data. Godot also provide some collection types.
+
+The main difference between the `.NET collections <https://learn.microsoft.com/en-us/dotnet/standard/collections/>`_
+and the Godot collections is that the .NET collections are implemented in C# while
+the Godot collections are implemented in C++ and the Godot C# API is a wrapper over it,
+this is an important distinction since it means every operation on a Godot collection
+requires marshaling which can be expensive specially inside in a loop.
+
+Due to the performance implications, using a Godot collection is usually not the
```

Comment:
I feel like this sentence is a bit long and hard to read. Maybe cutting it in halves would be good.
> Due to the performance implications, using Godot collections is only recommended if you need to interact with a Godot API. Since Godot only understands its own collection types, it's required to use them when talking to the engine.

Reply:
How about this?

```
Due to the performance implications, using Godot collections is only recommended when absolutely necessary (such as interacting with the Godot API). Godot only understands its own collection types, so it's required to use them when talking to the engine.
```

Reply:
Sounds good to me.